packages/expo-dev-launcher/android/src/main/java/com/facebook/react/devsupport/DevLauncherSettings.kt

```diff
@@ -0,0 +1,85 @@
+package com.facebook.react.devsupport
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.content.SharedPreferences.OnSharedPreferenceChangeListener
+import android.preference.PreferenceManager
+import com.facebook.react.common.build.ReactBuildConfig
+import com.facebook.react.devsupport.DevLauncherSettings.Listener
+import com.facebook.react.modules.debug.interfaces.DeveloperSettings
+import com.facebook.react.packagerconnection.PackagerConnectionSettings
+import expo.modules.devlauncher.react.DevLauncherPackagerConnectionSettings
+
+/**
+ * Implementation has been copied from reacts internal [DevInternalSettings] class
+ */
+internal class DevLauncherSettings(
+  context: Context,
+  debugServerHost: String,
+  private val listener: Listener? = Listener {}
+) : DeveloperSettings, OnSharedPreferenceChangeListener {
```

Comment:
i would recommend to have a shared open class `DevClientDevSettingsBase` that both DevLauncherSettings and DevMenuDevSettings can inherit from. then we could better differentiate what code did we change from upstream. it would also help when we need to sync changes from upstream. the idea is same as the `NonFinalBridgeDevSupportManager`.

Reply:
Updated 👍